tools/src/lib.rs

```diff
@@ -79,90 +145,197 @@ pub mod headers {
 
         let base_header_content = read(&mut BufReader::new(&base_header));
         let base_defs = collect_definitions(&base_header_content);
-        let base_defs_set: HashSet<_> = base_defs.iter().map(Match::as_str).collect();
+        let base_defs_set: HashSet<_> = base_defs.iter().map(|s| s.str).collect();
 
-        let mut base_new_parts = vec![&base_header_content[..base_defs.last().unwrap().end()]];
+        let mut base_new_parts = vec![&base_header_content[..base_defs.last().unwrap().end]];
         for child_def in &unique_child_defs {
             if base_defs_set.contains(child_def.as_str()) {
                 continue;
             }
             base_new_parts.push(child_def);
         }
-        base_new_parts.push(&base_header_content[base_defs.last().unwrap().end()..]);
+        base_new_parts.push(&base_header_content[base_defs.last().unwrap().end..]);
```

Comment:
non-blocking general comment about unwrap() usage: I think we'd like to expand the use of the tools crate to help facilitate building libdatadog in the SDKs. We probably want to at least switch to expect to make debugging build failures easier. @hoolioh maybe this is something we can do as part of the publishing and versioning project?